applications/GeoMechanicsApplication/custom_elements/geo_steady_state_Pw_piping_element.h

```diff
@@ -140,6 +143,80 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoSteadyStatePwPipingElement : publ
                std::abs(head_gradient);
     }
 
+    void CalculateOnIntegrationPoints(const Variable<bool>& rVariable,
+                                      std::vector<bool>&    rValues,
+                                      const ProcessInfo&    rCurrentProcessInfo) override
+    {
+        if (rVariable == PIPE_ACTIVE) {
+            const auto number_of_integration_points =
+                this->GetGeometry().IntegrationPointsNumber(this->GetIntegrationMethod());
+            rValues.resize(number_of_integration_points);
+            std::fill_n(rValues.begin(), number_of_integration_points, this->GetValue(rVariable));
+        }
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<double>& rVariable,
+                                      std::vector<double>&    rValues,
+                                      const ProcessInfo&      rCurrentProcessInfo) override
+    {
+        if (rVariable == PIPE_HEIGHT) {
+            const auto number_of_integration_points =
+                this->GetGeometry().IntegrationPointsNumber(this->GetIntegrationMethod());
+            rValues.resize(number_of_integration_points);
+            std::fill_n(rValues.begin(), number_of_integration_points, this->GetValue(rVariable));
+        }
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<array_1d<double, 3>>& rVariable,
+                                      std::vector<array_1d<double, 3>>&    rValues,
+                                      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (rVariable == FLUID_FLUX_VECTOR || rVariable == LOCAL_FLUID_FLUX_VECTOR) {
+            const auto number_of_integration_points =
+                this->GetGeometry().IntegrationPointsNumber(this->GetIntegrationMethod());
+            rValues.resize(number_of_integration_points);
+            auto        dynamic_viscosity_inverse = 1.0 / this->GetProperties()[DYNAMIC_VISCOSITY];
+            auto        constitutive_matrix = FillPermeabilityMatrix(this->GetValue(PIPE_HEIGHT));
```

Comment:
These two local variables can be made `const`:
```suggestion
            const auto  dynamic_viscosity_inverse = 1.0 / this->GetProperties()[DYNAMIC_VISCOSITY];
            const auto  constitutive_matrix = FillPermeabilityMatrix(this->GetValue(PIPE_HEIGHT));
```

Reply:
done